kb/compiling-and-building/using-mbedtls-in-microsoft-visual-studio-2015.md

```diff
@@ -0,0 +1,519 @@
+# Arm Mbed TLS in Microsoft Visual Studio 2015
+
+This tutorial shows how to get started with the Mbed TLS cryptography library in a Windows environment, using Microsoft Visual Studio 2015. This tutorial uses the sample client application (from [this example](https://tls.mbed.org/kb/how-to/mbedtls-tutorial)). This application sends an HTTP request to read an HTML page from a server.This tutorial uses Mbed TLS to enable encrypting our communication with the server with TLS. Note that Mbed TLS supports Visual Studio starting from version 2010.
+
+There are two possible ways to build Mbed TLS on Windows:
+
+- Using the solution file supplied with the Mbed TLS source code.
+- Using `cmake` to generate a Visual Studio solution file.
+
+This step-by-step tutorial shows you how to set up Visual Studio for Mbed TLS and the sample application. It assumes that you have already installed Visual Studio for C++. If generating a solution file using `cmake`, you should also install CMake. We tested this tutorial with `cmake` version 3.10.1.
+
+The sample application is a command-line program, written in C. You can just as easily integrate Mbed TLS in any C or C++ application, with or without a (graphical) user interface.
+
+## Download Mbed TLS
+
+First, you need to get the Mbed TLS source code. Download `mbedtls-<version\>-gpl.tgz` from [the Mbed TLS dowload location](https://tls.mbed.org/download). To unpack this file, you can use a tool like [7-zip](http://www.7-zip.org/).
+
+The default installation of Visual Studio creates this folder: My Documents\Visual Studio 2015\Projects. Extract the contents of the .tgz file to this location. Make sure to extract the second occurrence of the `mbedtls-<version>` folder. Otherwise, the paths given later in this tutorial won't match.
+
+<span class="notes">**Note:** You can extract the contents to any location you want, but for the purpose of this tutorial, the download location is the Visual Studio projects folder.</span>
+
+## Use the solution file supplied with the Mbed TLS source code
+
+### Open the Visual Studio solution
+
+You now have this folder: My Documents\Visual Studio 2015\Projects\mbedtls-<version\>\visualc\VS2010.
+
+In this folder, you can find the file `mbedTLS.sln`. This is a Visual Studio `solution` file, which holds all the components Mbed TLS consists of and the rules to build the Mbed TLS library.
+
+<span class="images">![](https://mbed-tls-docs-images.s3.amazonaws.com/MbedTLS-Tutorial-VS2015-1.png)</span>
```

Comment:
The images in this file (and a few in other files) have absolute URLs which are broken, and the corresponding images are not in this pull request.